src/shared/pretty-print.c

```diff
@@ -228,58 +213,78 @@ static int cat_file(const char *filename, bool newline, CatFlags flags) {
                         break;
 
-                LineType line_type = classify_line_type(line, flags);
-                if (FLAGS_SET(flags, CAT_TLDR)) {
-                        if (line_type == LINE_SECTION) {
-                                /* The start of a section, let's not print it yet. */
+                const char *l = skip_leading_chars(line, WHITESPACE);
+
+                /* comment */
+                if (*l != '\0' && strchr(COMMENTS, *l)) {
+                        if (!FLAGS_SET(flags, CAT_TLDR))
+                                printf("%s%s%s\n", ansi_highlight_grey(), line, ansi_normal());
+                        continue;
+                }
+
+                /* empty line */
+                if (isempty(l) && FLAGS_SET(flags, CAT_TLDR) && !continued)
+                        continue;
+
+                /* section */
+                if (FLAGS_SET(flags, CAT_FORMAT_HAS_SECTIONS) && *l == '[' && !continued) {
+                        if (FLAGS_SET(flags, CAT_TLDR))
+                                /* On TLDR, let's not print it yet. */
                                 free_and_replace(section, line);
-                                continue;
-                        }
+                        else
+                                printf("%s%s%s\n", ansi_highlight_cyan(), line, ansi_normal());
+                        continue;
+                }
 
-                        if (line_type == LINE_COMMENT)
-                                continue;
+                /* normal line */
 
-                        /* Before we print the actual line, print the last section header */
-                        if (section) {
-                                /* Do not print redundant section headers */
-                                if (!streq_ptr(section, old_section))
-                                        printf("%s%s%s\n",
-                                               ansi_highlight_cyan(),
-                                               section,
-                                               ansi_normal());
+                /* Before we print the line, print the last section header. */
+                if (FLAGS_SET(flags, CAT_TLDR) && section) {
+                        /* Do not print redundant section headers */
+                        if (!streq_ptr(section, old_section))
+                                printf("%s%s%s\n", ansi_highlight_cyan(), section, ansi_normal());
 
-                                free_and_replace(old_section, section);
-                        }
+                        free_and_replace(old_section, section);
+                }
+
+                /* Check if the line ends with a backslash. */
+                bool escaped = false;
+                char *e;
+                for (e = line; *e != '\0'; e++) {
+                        if (escaped)
+                                escaped = false;
+                        else if (*e == '\\')
+                                escaped = true;
```

Comment:
Hmm? Why not strrchr and check if the next char is NUL?

Reply:
"\\\\" (two backslashes) at the end means one backslash.